### PR TITLE
Actions dynamic fields

### PR DESCRIPTION
**Pull request by nick-Ag**
A series of improvements to the dynamic fields feature. Dynamic fields allow builders to fetch predefined choices for input fields from a downstream destination. This feature was previously implemented but never enabled. 

This PR improves the builder experience for these fields, updates the return types for the functions defined, and introduces a test utility to allow builders to write unit tests for their dynamic fields.

Fab-5-engine PR: https://github.com/segmentio/fab-5-engine/pull/467

### Unit Test Utility
Tested the new unit test utility with this tester dynamic field: https://github.com/segmentio/action-destinations/pull/683/files#diff-082598c5300da4aabe4fdbd91e2ffec769fdba33c6910ee3a9c6bde3006d1d3dR278

Unit test passes correctly and the utility passes back responses as expected.

### Stage Testing
Tested successfully in stage using this test code: https://github.com/segmentio/action-destinations/pull/679/commits/360fc6177a25901fa835bf4802137ca2e4ef0e69#diff-6f4cad74d0feada0d69726ad7aaf70acfa49d56032f96ad1d38589ea4f0095e1R37
The test dynamic field fetched from a Cloudflare worker which returned a simple string.

Screen Shot 2022-07-14 at 4 17 35 PM

### Stress Test
I was interested in testing our UI in cases where large numbers of items are returned.
Results:

100 Items: Works well, no visible perf impact. UI looks good.
Screen Shot 2022-07-15 at 10 08 55 AM

500 items: All looks good.
Screen Shot 2022-07-15 at 10 09 21 AM

1000 items: Starts to encounter some lag. 1.3s Response time.
Screen Shot 2022-07-15 at 10 10 14 AM

3000 items: UI looks good but considerable lag. 4.3s response.
Screen Shot 2022-07-15 at 10 10 46 AM

5000 items: Starts to break down. Timedout and only 3000 items were displayed in UI.
Screen Shot 2022-07-15 at 10 11 47 AM

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/core/src/destination-kit/index.ts [hunk 1]
```diff
@@ -369,6 +369,19 @@ export class Destination<Settings = JSONObject> {
     return [{ output: 'successfully processed batch of events' }]
   }
 
+  public async executeDynamicField(
```
Comment (dlasky) [inline packages/core/src/destination-kit/index.ts:372]: its so bizzare that this didn't exist?

**Comment by dlasky** [comment]
i do wonder if in app we shouldn't add something specific for error handling for this that implies the partner could be experiencing API issues?

File: packages/destination-actions/src/destinations/pipedrive/pipedriveApi/pipedrive-client.ts [hunk 2]
```diff
@@ -91,10 +91,8 @@ class PipedriveClient {
       value: f.key
     }))
     const record = {
-      body: {
-        data: fields,
-        pagination: {}
-      }
+      choices: fields,
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/pipedrive/pipedriveApi/pipedrive-client.ts:94]: I wonder if we should disable Pipedrive's dynamic field for now (via flagon) until we get approval from them to include it? cc @mericsson

**Comment by nick-Ag** [comment]
Will be deployed during the Wednesday deploy

**Pull request merged**